05.ls/ls.rb

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+MAX_COLUMN = 3
+SPACE_LENGTH = 3
+
+def main
+  files = search_files(ARGV[0])
+  output(files)
+end
+
+def search_files(path)
+  return [path] if FileTest.file?(path.to_s)
+
+  dirpath = ARGV.empty? ? Dir.getwd : path
+  files = []
+  Dir.foreach(dirpath) do |file|
+    files << file unless file.start_with?('.')
+  end
+  files.sort
+end
+
+def output(files_name)
+  row_count = (files_name.size.to_f / MAX_COLUMN).ceil
+  display_width = adjust_width(files_name).max
```

Comment:
```suggestion
  display_width = hogehoge(files_name)
```

maxを呼び出さなくてもいいメソッド設計がいいのでは？と思いました。

Reply:
メソッド内で最大値を取得して完結させた方が良い。という意味と解釈し、納得はしているのですが自分の中ではこれ以上の方法が見つかりませんでした…

mapメソッドの代わりにmax_byメソッドを使おうとしたのですが、これだと最大長の文字列で返してしまうので数値で返すためには二回`count_chars`メソッドを呼び出すことになるため冗長になってしまい、それなら今のままの方がスマートという結論になりました。
他にもっと良いアプローチ方法があればアドバイスいただけると幸いです。